src/go/MONIT/intelligence.go

```diff
@@ -0,0 +1,320 @@
+func silenceMaintenance(filteredAlerts amData) {
+
+	if configJSON.Verbose > 1 {
+		if len(filteredAlerts.Data) == 0 {
+			log.Printf("No Maintenance Alert Found")
+			return
+		}
+	}
+
+	ifAnyAlert := false
+
+	for _, each := range filteredAlerts.Data {
+		for k, v := range each.Labels {
+			if k == configJSON.SearchingLabel {
+				for _, ins := range strings.Split(v.(string), ",") {
+					if val, ok := silenceMap[ins]; ok {
+						ifAnyAlert = true
+						silence(val, each)
+					}
+				}
+			}
+		}
+	}
+
+	if configJSON.Verbose > 1 {
+		if ifAnyAlert == false {
+			log.Printf("No Alert found for Silencing")
+		}
+	}
+}
+
+//Function for filtering maintenance alerts
+func filterMaintenance(amdata amData) amData {
+	silenceMap = make(map[string]amJSON)
+	var maintenanceData amData
+
+	for _, each := range amdata.Data {
+		for k, v := range each.Labels {
+			if k == "severity" && v == configJSON.SeverityFilter {
+				maintenanceData.Data = append(maintenanceData.Data, each)
+			}
+			if k == configJSON.SearchingLabel {
+				if len(strings.Split(v.(string), ",")) == 1 {
+					silenceMap[v.(string)] = each
+				}
+			}
+		}
+	}
+	return maintenanceData
+}
+
+//Function running all logics
+func run() {
+
+	var amdata amData
+	get(&amdata)
+
+	filtered := filterMaintenance(amdata)
+	silenceMaintenance(filtered)
+
+}
+
+func parseConfig(verbose int) {
+
+	configFile := os.Getenv("INTELLIGENCE_CONFIG_PATH") //INTELLIGENCE_CONFIG_PATH Environment Variable storing config filepath.
+
+	//Defaults in case no config file is provided
+	configJSON.CMSMONURL = "https://cms-monitoring.cern.ch"
+	configJSON.GetAlertsAPI = "/api/v1/alerts?active=true&silenced=false&inhibited=false&unprocessed=false"
+	configJSON.PostSilenceAPI = "/api/v1/silences"
+	configJSON.SeverityFilter = "maintenance"
+	configJSON.SearchingLabel = "instance"
+	configJSON.UniqueLabel = "alertname"
+	configJSON.Comment = "maintenance"
+	configJSON.CreatedBy = "admin"
+	configJSON.HttpTimeout = 3 //3 secs timeout for HTTP requests
+	configJSON.Interval = 10   // 10 sec interval for the service
+	configJSON.Verbose = verbose
+
+	if configFile != "" {
+		jsonFile, e := os.Open(configFile)
+		if e != nil {
+			log.Fatalf("Config File not found, error: %s", e)
+		}
+		defer jsonFile.Close()
+		decoder := json.NewDecoder(jsonFile)
+		err := decoder.Decode(&configJSON)
+		if err != nil {
+			log.Fatalf("Config JSON File can't be loaded, error: %s", err)
+		} else if configJSON.Verbose > 0 {
+			log.Printf("Load config from %s\n", configFile)
+		}
+	}
+
+	if configJSON.Verbose > 0 {
+		log.SetFlags(log.LstdFlags | log.Lshortfile)
+	} else {
+		log.SetFlags(log.LstdFlags)
+	}
+
+	if configJSON.Verbose > 1 {
+		log.Printf("Configuration:\n%+v\n", configJSON)
+	}
+
+}
+
+func runInfinite() {
+	for true {
+		run()
+		time.Sleep(configJSON.Interval * time.Second)
+	}
+}
+
+func main() {
+
+	var verbose int
+	flag.IntVar(&verbose, "verbose", 0, "Verbosity Level, can be overwritten in config")
```

Comment:
make config option to pass config file which will replace reading it from environment

Reply:
Valentin,
When we create a systemd service it will be helpful if we keep Environment variable. We can easily pass envs to sytemd service file like this.

`Environment="INTELLIGENCE_CONFIG_PATH"`

Reply:
we're fully migrating to k8s, therefore there is no need for systems as we'll most likely run it as service

Reply:
Okay.